config/lib/seed.js

```diff
@@ -92,3 +65,85 @@ function reportError(err) {
   console.log(err);
   console.log();
 }
+
+function performSeedUser(user) {
+
+  var User = mongoose.model('User');
+
+  return new Promise(function(resolve, reject) {
+
+    var seedUser = {
+      username: 'user',
+      password: 'User_Password1!',
+      provider: 'local',
+      email: 'user@localhost.com',
+      firstName: 'User',
+      lastName: 'Local',
+      displayName: 'User Local',
+      roles: ['user']
+    };
+
+    var seedAdmin = {
+      username: 'admin',
+      password: 'Admin_Password1!',
+      provider: 'local',
+      email: 'admin@localhost.com',
+      firstName: 'Admin',
+      lastName: 'Local',
+      displayName: 'Admin Local',
+      roles: ['user', 'admin']
+    };
+
+    var user = null;
+
+    //If production only seed admin if it does not exist
+    if (process.env.NODE_ENV === 'production') {
+      //Add Local Admin
+
+      user = new User(seedAdmin);
+
+      User.generateRandomPassphrase()
```

Comment:
I commented on the use of the closures here. So I figured I'd clarify here, since it's related to @ilanbiala comment...

@lirantal Here's an example of what I meant... 
```
User.generateRandomPassphrase()
.then(saveUser(user))
.then(completeSeedUser) // you can resolve your local promise in this function
.catch(reportError);

// saveUser would look like this
function saveUser (user) {
  return function (password) { // this is what I meant about using a closure, rather than a function
    ........
  }
}
```
Essentially, what you have is the same & it works the same. Maybe it's just a preference. However, it may make the code more readable & you can put all logic into their own functions & closures. For instance, if you return the user with the resolve from `saveUser` you could take the success reporting out of `saveUser` & into their own function where you only have to manage that code in one place.